### PR TITLE
Issue #262 primod rchsvatmapping layer dim

### PR DESCRIPTION
**Pull request by JoerivanEngelen**
This PR fixes issue #262, as well as adds some unit tests to verify that most common cases in which recharge can be provided work.

- 2D grid (y, x) with ``layer`` coord (succeeds)
- 3D grid with layer dimension, length 1 (succeeds)
- 3D grid with multiple layers (fails)
- 2D transient grid (fails)

I placed the new tests in the ``tests/test_mapping.py``, but I noticed there is a ``imod_coupler/mapping`` namespace, as well as a ``primod/mapping`` namespace. I was a bit in doubt whether mixing these in ``tests/test_mapping.py`` is a good idea. If you have ideas for a better location to put these tests, let me know.

### CODE REVIEW

**Comment by JoerivanEngelen** [comment]
I see some tests are labeled as "flaky", making it appear as if this is not my wrongdoing. Is this the case or is it likely I introduced errors in RibaMod?

**Review by Hofer-Julian — APPROVED** [review]
I am not super familiar with that part of the codebase, but the changes look good to me.

The testbench was red, but I think that was just because of a version mismatch between Ribasim and Ribasim Python. If it is green now, we can merge this from my side.

File: pre-processing/primod/mapping/rch_svat_mapping.py [hunk 1]
```diff
@@ -39,12 +39,13 @@ def __init__(self, svat: xr.DataArray, recharge: mf6.Recharge):
         super().__init__()
         self.dataset["svat"] = svat
         self.dataset["layer"] = xr.full_like(svat, 1)
-        if "layer" in recharge.dataset.coords:
-            self.dataset["rch_active"] = (
-                recharge.dataset["rate"].drop_vars("layer").notnull()
-            )
+        rate = recharge.dataset["rate"]
+        if "layer" in rate.dims:
+            rate_no_layer = rate.squeeze("layer", drop=True)
```
Comment (luitjansl) [inline pre-processing/primod/mapping/rch_svat_mapping.py:44]: should there be a check that the length of the layer dimension is 1?

File: tests/fixtures/fixture_modflow.py [hunk 2]
```diff
@@ -65,6 +65,15 @@ def make_mf6_simulation(gwf_model: mf6.GroundwaterFlowModel) -> mf6.Modflow6Simu
     return simulation
 
 
+def make_recharge(idomain: xr.DataArray) -> mf6.Recharge:
```
Comment (luitjansl) [inline tests/fixtures/fixture_modflow.py:68]: make_recharge_package?

File: tests/fixtures/fixture_modflow.py [hunk 3]
```diff
@@ -65,6 +65,15 @@ def make_mf6_simulation(gwf_model: mf6.GroundwaterFlowModel) -> mf6.Modflow6Simu
...
+    idomain_l1 = idomain.sel(layer=1)
+    recharge = xr.zeros_like(idomain_l1, dtype=float)
+    recharge[:, 0] = np.nan
```
Comment (luitjansl) [inline tests/fixtures/fixture_modflow.py:71]: are you assigning nan to all x for y =0? (if so, why is that? if something else, what exactly?)

**Review by luitjansl — APPROVED** [review]
approved (but some comments/questions)

File: pre-processing/primod/mapping/rch_svat_mapping.py (hunk 1 above)
Comment (JoerivanEngelen) [reply pre-processing/primod/mapping/rch_svat_mapping.py:44]: The squeeze method automatically throws an error if length is more than 1.

File: tests/fixtures/fixture_modflow.py (hunk 3 above)
Comment (JoerivanEngelen) [reply tests/fixtures/fixture_modflow.py:71]: The opposite, all cells in the first column (x) are assigned ``np.nan``. This is consistent with the MetaSWAP model, where these cells are inactive, and thus no mapping should take place to these cells.

Comment (JoerivanEngelen) [reply tests/fixtures/fixture_modflow.py:72]: I added a comment to clarify

File: tests/fixtures/fixture_modflow.py (hunk 2 above)
Comment (JoerivanEngelen) [reply tests/fixtures/fixture_modflow.py]: Renamed

**Pull request merged**